mlflow/tracing/display/display_handler.py

```diff
@@ -1,16 +1,66 @@
 import json
 import logging
 from typing import TYPE_CHECKING
+from urllib.parse import urlencode, urljoin
 
+import mlflow
 from mlflow.environment_variables import MLFLOW_MAX_TRACES_TO_DISPLAY_IN_NOTEBOOK
 from mlflow.utils.databricks_utils import is_in_databricks_runtime
+from mlflow.utils.uri import is_http_uri
 
 _logger = logging.getLogger(__name__)
 
 if TYPE_CHECKING:
     from mlflow.entities import Trace
 
 
+TRACE_RENDERER_ASSET_PATH = "/static-files/lib/notebook-trace-renderer/index.html"
+
+IFRAME_HTML = """
+<div>
+  <style scoped>
+  button {{
+    border: none;
+    border-radius: 4px;
+    background-color: rgb(34, 114, 180);
+    font-family: -apple-system, "system-ui", "Segoe UI", Roboto, "Helvetica Neue", Arial;
+    font-size: 13px;
+    color: white;
+    margin-top: 8px;
+    margin-bottom: 8px;
+    padding: 8px 16px;
+    cursor: pointer;
+  }}
+  button:hover {{
+    background-color: rgb(66, 153, 224);
+  }}
+  </style>
+  <button
+    onclick="const display = this.nextElementSibling.style.display;
+const isCollapsed = display === 'none';
+this.nextElementSibling.style.display = isCollapsed ? null : 'none';
+const verb = isCollapsed ? 'Collapse' : 'Expand';
+this.innerText = `${{verb}} MLflow Trace`;"
+  >Collapse MLflow Trace</button>
+  <iframe
+    id="trace-renderer"
+    style="width: 100%; height: 500px; border: none; resize: vertical;"
+    src="{src}"
+  />
+</div>
+"""
+
+
+def _get_notebook_iframe_html(traces: list["Trace"]):
+    # fetch assets from tracking server
+    uri = urljoin(mlflow.get_tracking_uri(), TRACE_RENDERER_ASSET_PATH)
+    query_string = _get_query_string_for_traces(traces)
+
+    # include mlflow version to invalidate cache when mlflow updates
+    src = f"{uri}?{query_string}&version={mlflow.__version__}"
```

Comment:
Is it possible to validate `src` is a valid URL to prevent template inejction?

Reply:
@harupy what kind of validations do you have in mind? like sending a test request and making sure the response is 200? to prevent template injection, how about running `html.escape()` on `uri`, since that's the only part under control by the user?

Reply:
for valid URLs (e.g. `http://localhost:5000/static-files/lib/notebook-trace-renderer/index.html`), `html.escape()` should return it unchanged. for malicious ones (like `http://example.com\" <my-malicious-element /> \"`), it will escape all the special characters like `<` and `"`

Reply:
oh wait actually i guess there's no harm to run it for the entire `src`. i guess users could potentially control the trace ID and stuff as well